File: client/src/boundary/model/BoundaryGameStats.ts
export default class BoundaryGameStats {
  readonly hitsRemaining: number;
  readonly shipsDestroyed: number;

  constructor(hitsRemaining: number, shipsDestroyed: number) {
    this.hitsRemaining = hitsRemaining;
    this.shipsDestroyed = shipsDestroyed;
  }
}

File: server/battleships/use_cases/api/src/main/java/eu/optas/use_cases/api/BoundaryGameStats.java
package eu.optas.use_cases.api;

public class BoundaryGameStats {
    private final int hitsRemaining;
    private final int shipsDestroyed;

    public BoundaryGameStats(int hitsRemaining, int shipsDestroyed) {
        this.hitsRemaining = hitsRemaining;
        this.shipsDestroyed = shipsDestroyed;
    }

    public int getHitsRemaining() {
        return hitsRemaining;
    }

    public int getShipsDestroyed() {
        return shipsDestroyed;
    }
}
